File: app/src/main/java/com/google/sample/cloudvision/searchpojos/SearchInformation.java

package com.google.sample.cloudvision.searchpojos;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class SearchInformation implements Parcelable
{

    @SerializedName("searchTime")
    @Expose
    private Double searchTime;
    @SerializedName("formattedSearchTime")
    @Expose
    private String formattedSearchTime;
    @SerializedName("totalResults")
    @Expose
    private String totalResults;
    @SerializedName("formattedTotalResults")
    @Expose
    private String formattedTotalResults;
    public final static Creator<SearchInformation> CREATOR = new Creator<SearchInformation>() {


        @SuppressWarnings({
            "unchecked"
        })
        public SearchInformation createFromParcel(Parcel in) {
            SearchInformation instance = new SearchInformation();
            instance.searchTime = ((Double) in.readValue((Double.class.getClassLoader())));
            instance.formattedSearchTime = ((String) in.readValue((String.class.getClassLoader())));
            instance.totalResults = ((String) in.readValue((String.class.getClassLoader())));
            instance.formattedTotalResults = ((String) in.readValue((String.class.getClassLoader())));
            return instance;
        }

        public SearchInformation[] newArray(int size) {
            return (new SearchInformation[size]);
        }

    }
    ;

    public Double getSearchTime() {
        return searchTime;
    }

    public void setSearchTime(Double searchTime) {
        this.searchTime = searchTime;
    }

    public String getFormattedSearchTime() {
        return formattedSearchTime;
    }

    public void setFormattedSearchTime(String formattedSearchTime) {
        this.formattedSearchTime = formattedSearchTime;
    }

    public String getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(String totalResults) {
        this.totalResults = totalResults;
    }

    public String getFormattedTotalResults() {
        return formattedTotalResults;
    }

    public void setFormattedTotalResults(String formattedTotalResults) {
        this.formattedTotalResults = formattedTotalResults;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(searchTime);
        dest.writeValue(formattedSearchTime);
        dest.writeValue(totalResults);
        dest.writeValue(formattedTotalResults);
    }

    public int describeContents() {
        return  0;
    }

}

File: app/src/main/java/com/google/sample/cloudvision/searchpojos/SearchInformation.kt
package com.google.sample.cloudvision.searchpojos

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class SearchInformation : Parcelable {

    @SerializedName("searchTime")
    @Expose
    var searchTime: Double? = null
    @SerializedName("formattedSearchTime")
    @Expose
    var formattedSearchTime: String? = null
    @SerializedName("totalResults")
    @Expose
    var totalResults: String? = null
    @SerializedName("formattedTotalResults")
    @Expose
    var formattedTotalResults: String? = null

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(searchTime)
        dest.writeValue(formattedSearchTime)
        dest.writeValue(totalResults)
        dest.writeValue(formattedTotalResults)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<SearchInformation> = object : Parcelable.Creator<SearchInformation> {


            override fun createFromParcel(`in`: Parcel): SearchInformation {
                val instance = SearchInformation()
                instance.searchTime = `in`.readValue(Double::class.java.classLoader) as Double
                instance.formattedSearchTime = `in`.readValue(String::class.java.classLoader) as String
                instance.totalResults = `in`.readValue(String::class.java.classLoader) as String
                instance.formattedTotalResults = `in`.readValue(String::class.java.classLoader) as String
                return instance
            }

            override fun newArray(size: Int): Array<SearchInformation?> {
                return arrayOfNulls(size)
            }

        }
    }

}
